src/wallet/wallet.cpp

```diff
@@ -3811,6 +3812,18 @@ std::vector<std::string> CWallet::GetDestValues(const std::string& prefix) const
     return values;
 }
 
+bool CWallet::SetAddressReceiveRequest(WalletBatch& batch, const CTxDestination& dest, const std::string& id, const std::string& value)
+{
+    const std::string key{"rr" + id}; // "rr" prefix = "receive request" in destdata
+    CAddressBookData& data = m_address_book.at(dest);
+    if (value.empty()) {
+        return data.destdata.erase(key) && batch.EraseDestData(EncodeDestination(dest), key);
+    } else {
+        data.destdata[key] = value;
```

Comment:
dd8a9d019aa449f3a6c5ad989cb6bdd9f03d1306

Should update `destdata` iff `WriteDestData` succeeds?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/21353#discussion_r588890661

> [dd8a9d0](https://github.com/bitcoin/bitcoin/commit/dd8a9d019aa449f3a6c5ad989cb6bdd9f03d1306)
> 
> Should update `destdata` iff `WriteDestData` succeeds?

Thanks! Implemented